packages/node-experimental/src/sdk/scope.ts

```diff
@@ -0,0 +1,138 @@
+import { dateTimestampInSeconds } from '@sentry/utils';
+
+import { getActiveSpan } from './trace';
+
+const DEFAULT_MAX_BREADCRUMBS = 100;
+
+/**
+ * This is a fork of the base Transaction with OTEL specific stuff added.
+ * Note that we do not solve this via an actual subclass, but by wrapping this in a proxy when we need it -
+ * as we can't easily control all the places a transaction may be created.
+ */
+interface TransactionWithBreadcrumbs extends Transaction {
+  _breadcrumbs: Breadcrumb[];
+
+  /** Get all breadcrumbs added to this transaction. */
+  getBreadcrumbs(): Breadcrumb[];
+
+  /** Add a breadcrumb to this transaction. */
+  addBreadcrumb(breadcrumb: Breadcrumb, maxBreadcrumbs?: number): void;
+}
+
+/** A fork of the classic scope with some otel specific stuff. */
+export class OtelScope extends Scope {
+  /**
+   * @inheritDoc
+   */
+  public static clone(scope?: Scope): Scope {
+    const newScope = new OtelScope();
+    if (scope) {
+      newScope._breadcrumbs = [...scope['_breadcrumbs']];
+      newScope._tags = { ...scope['_tags'] };
+      newScope._extra = { ...scope['_extra'] };
+      newScope._contexts = { ...scope['_contexts'] };
+      newScope._user = scope['_user'];
+      newScope._level = scope['_level'];
+      newScope._span = scope['_span'];
+      newScope._session = scope['_session'];
+      newScope._transactionName = scope['_transactionName'];
+      newScope._fingerprint = scope['_fingerprint'];
+      newScope._eventProcessors = [...scope['_eventProcessors']];
+      newScope._requestSession = scope['_requestSession'];
+      newScope._attachments = [...scope['_attachments']];
+      newScope._sdkProcessingMetadata = { ...scope['_sdkProcessingMetadata'] };
+      newScope._propagationContext = { ...scope['_propagationContext'] };
+    }
+    return newScope;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public addBreadcrumb(breadcrumb: Breadcrumb, maxBreadcrumbs?: number): this {
+    const transaction = getActiveTransaction();
+
+    if (transaction) {
+      transaction.addBreadcrumb(breadcrumb, maxBreadcrumbs);
+      return this;
+    }
+
+    return super.addBreadcrumb(breadcrumb, maxBreadcrumbs);
+  }
+
+  /**
+   * @inheritDoc
+   */
+  protected _getBreadcrumbs(): Breadcrumb[] {
+    const transaction = getActiveTransaction();
+    const transactionBreadcrumbs = transaction ? transaction.getBreadcrumbs() : [];
+
+    return this._breadcrumbs.concat(transactionBreadcrumbs);
+  }
+}
+
+/**
+ * This gets the currently active transaction,
+ * and ensures to wrap it so that we can store breadcrumbs on it.
+ */
+function getActiveTransaction(): TransactionWithBreadcrumbs | undefined {
+  const activeSpan = getActiveSpan();
+  const transaction = activeSpan && activeSpan.transaction;
+
+  if (!transaction) {
+    return undefined;
+  }
+
+  if (transactionHasBreadcrumbs(transaction)) {
+    return transaction;
+  }
+
+  return new Proxy(transaction as TransactionWithBreadcrumbs, {
```

Comment:
l: we can proxy patch `startTransaction` (hub extension) so that we don't have to rely on `getActiveTransaction` being used all the time.

Can skip this for now just to get it merged.

Reply:
Good idea, wrote something up here: https://github.com/getsentry/sentry-javascript/pull/9010